src/components/Select/Option.vue

```diff
@@ -1,87 +1,115 @@
+      </template>
+      <template v-slot:label="scope">
+        <slot name="label" v-bind="scope" />
+      </template>
+    </OptionLabel>
+    <template v-if="isSelected">
+      <i class="pi pi-check-line pi-lg" />
+    </template>
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue'
+import { Vue, prop } from 'vue-class-component'
+import { Component } from '@/utilities/vue-class-component'
+import OptionLabel from '@/components/Select/OptionLabel.vue'
 
-export default defineComponent({
-  name: 'Option',
-  emits: {
-    click(e: Event, ...args: any[]) {
-      return { e, ...args }
-    }
-  },
-  props: {
-    value: {
-      type: String,
-      required: true,
-      default: 'Option'
-    },
-    data: {
-      type: [String, Object]
-    },
-    icon: {
-      type: String,
-      default: ''
-    },
-    selected: {
-      type: Boolean,
-      default: false
-    },
-    disabled: {
-      type: Boolean,
-      default: false
-    }
-  },
-  data() {
+class Props {
+  label = prop<string>({ required: true })
+  value = prop({ required: true })
+  icon = prop<string>({ default: '' })
+  disabled = prop<boolean | string>({ default: false })
+  selected = prop<boolean | string>({ default: false })
+}
+
+@Component({
+  emits: ['select'],
+  components: {
+    OptionLabel
+  }
+})
+export default class Option extends Vue.with(Props) {
+
+  get isDisabled() {
+    return this.disabled || this.disabled === ''
+  }
+
+  get isSelected() {
+    return this.selected || this.selected === ''
+  }
+
+  get scope() {
     return {
-      hovered: false
+      label: this.label,
+      value: this.value,
+      icon: this.icon,
+      disabled: this.isDisabled,
+      selected: this.isSelected
     }
-  },
-  computed: {
-    iconClass(): string | null {
-      return this.icon.length > 0 ? `pi-${this.icon}` : null
+  }
+
+  get classes() {
+    return {
+      option: {
+        'option--disabled': this.isDisabled,
+        'option--selected': this.isSelected
+      },
+      icon: `pi pi-1x mr-1 pi-${this.icon}`
     }
-  },
-  methods: {
-    choose(e: Event): Event {
-      this.$emit('click', e, this.value, this.icon, this.data)
-      return e
-    },
-
-    handleFocus(): void {
-      if (this.disabled) return
-      this.hovered = true
-    },
-
-    handleBlur(): void {
-      this.hovered = false
+  }
+
+  private click() {
+    if(this.isDisabled) {
+      return
     }
+
+    this.$emit('select', this.value)
   }
-})
+
+}
 </script>
 
-<style lang="scss" scoped>
-@use '../../styles/components/select';
-</style>
+<style lang="scss">
+@use '../../styles/abstracts/variables';
+
+.option {
+  height: var(--select-height, 58px);
+  color: #{variables.$text--primary};
+  display: flex;
+  align-items: center;
+  padding: 0 20px;
+  cursor: pointer;
+  user-select: none;
+
+  &:hover {
+    background-color: #{variables.$grey-10};
+  }
+
+  &:not(:last-child) {
+    border-bottom: 1px solid #f7f7f8;
```

Comment:
seems weird to have a specific white here that's not from our variables?

Reply:
updated with an existing variable.